dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1741,6 +1741,12 @@
 		</ResolveNativeReferences>
 	</Target>
 
+	<Target Name="_ValidateOutputType"
+		Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst' "
+		BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
```

Comment:
One of the issues I found was these targets where run after the error was triggered. I will with the one you propose

Reply:
It can't be added to any of the targets defined here because the error happens before:

<img width="1221" alt="image" src="https://user-images.githubusercontent.com/111134/203376006-188928fb-a683-4227-a39d-fcd161d145f2.png">

Reply:
> It can't be linked to any of the targets defined here because the error happens before:

Good point.

I think you can add yourself to the BundleDependsOn property:

```xml
 <BundleDependsOn>
 	_ValidateOutputType;
 	$(BundleDependsOn); 
 </BundleDependsOn> 
```

Reply:
Thanks for the hint. It's working now :)

Reply:
I have issues with that approach, it fails with some projects. I am debugging it

Reply:
False positive, the Bundle target it still too late. I have only made it work earlier targets like: ProcessFrameworkReferences. I attached a screenshot of the targets that are run before the error

Reply:
Oh ops, `BundleDependsOn` is a typo, it's supposed to be `BuildDependsOn`:

```xml
 <BundleDependsOn>
 	_VerifyValidOutputType;
 	$(BundleDependsOn); 
 </BundleDependsOn> 
```

and that works as long as it's set *after* where we also set `BundleDependsOn` here:

https://github.com/xamarin/xamarin-macios/blob/e4c940cdb81050f5ea9b8793c90993682e115a43/dotnet/targets/Xamarin.Shared.Sdk.targets#L172-L211

basically the `_VerifyValidOutputType` target must be listed before the `_CollectBundleResources` target in `BuildDependsOn` here:

https://github.com/xamarin/xamarin-macios/blob/e4c940cdb81050f5ea9b8793c90993682e115a43/dotnet/targets/Xamarin.Shared.Sdk.targets#L199